tests/verify.rs

```diff
@@ -1,19 +1,23 @@
 use std::fs;
 
+use rstest::rstest;
+
 /// Path to test app
 const CRATE: &str = "test-flip-link-app";
 /// Example firmware in `$CRATE/examples`
 const FILES: [&str; 4] = ["crash", "exception", "hello", "panic"];
 /// Compilation target firmware is build for
 const TARGET: &str = "thumbv7m-none-eabi";
 
-#[test]
-fn should_link_example_firmware() -> anyhow::Result<()> {
+#[rstest]
+#[case::normal(true)]
+#[case::custom_linkerscript(false)]
+fn should_link_example_firmware(#[case] default_features: bool) -> anyhow::Result<()> {
```

Comment:
there's a potential race condition here: `rstest` produces *2* `#[test]` functions. These will run in parallel: meaning that 2 `cargo build` processes will run in parallel. Both processes will write an ELF file to `target/thumb/debug/elf`; the instance that takes longer will overwrite the file that the other instance wrote. That file will be open / inspected by the `#[test]` function. There's no guarantee that each `#[test]` function will open the *right* instance: more likely both `#[test]` functions will end up inspecting the same ELF file.

To solve the issue, the `#[test]` functions should run *sequentially*. there's two ways to do that:
- you can pass `--test-threads=1` to the `cargo test` invocation
- you can use a `Mutex` in this file as a semaphore to ensure only *one* `#[test]` function has access to the `target` directory at any time.

Given that these tests are run using the `cargo xtest` command, I think `--test-threads` is a good solution.

Reply:
Went forward with the `--test-threads` option in 72e74d33f6a67714a22d53a1b2dae60460430d34.